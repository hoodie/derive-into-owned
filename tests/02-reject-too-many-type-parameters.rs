/// this ought to not compile, because
use std::borrow::Cow;

use derive_into_owned::IntoOwned;

// TODO: this already doesn't compile, but the test's still try to compile it anyway
#[derive(IntoOwned)]
pub struct Parsed<'a, T> {
    content: Cow<'a, String>,
    stuff: T,
}

fn main() {}
