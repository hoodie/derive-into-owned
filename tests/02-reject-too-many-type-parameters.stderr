error: proc-macro derive panicked
 --> tests/02-reject-too-many-type-parameters.rs:7:10
  |
7 | #[derive(IntoOwned)]
  |          ^^^^^^^^^
  |
  = help: message: this type has too many type parameters, only a single lifetime parameter is supported
